### PR TITLE
Updated navigation to ignore coastal flowlines

### LINKED ISSUE #39: Navigation on coastal flowlines is unconstrained
This issue is a followup to the working group discussion back in October.

NHDPlus Navigation is constrained by cost (distance or time) taken from the flowline VAA attributes.  This logic dates to NHDPlus 1.0 days when all flowline VAA attributes had some kind of cost value.  

However when NHDPlus 2.0 was released, it included for the first time coastal flowlines - all of which have a distance cost of zero.  The NHDPlus team has verified this is as intended.

The problem occurs when for whatever reason a coastal flowline is selected for navigation.  For example a user submits a simple request to navigation "upstream" for 5 KM from coast flowline ComID 8379397 near Lusby on the Maryland coast of the Chesapeake Bay.
capture1

As coastal flowlines have a distance cost of zero the navigation will run the entire linked distance to the Mexican border
capture

As this result is probably undesired and as this result is onerous to the server and client, it might be best to disallow navigation from and upon coastal flowlines.

### PR DESCRIPTION
**Pull request by EthanGrahn**
Before making a pull request
----------------------------

- [x] Make sure all tests run
- [x] Update the changelog appropriately

Updated navigation to ignore coastal flowlines
-----------
Created an fcode check in `navigate.xml` to ignore coastal features (fcode == 56600).

Description
-----------
https://internal.cida.usgs.gov/jira/browse/NHGF-30 
Navigation requests will now stop once they reach a coastal feature. If the request starts on a coastal feature, only that feature will be returned. In order to test using the demo database you will need to create fake coasts. Run the following psql command against the database to create example coasts.
```
update nhdplus.plusflowlinevaa_np21 set fcode = 56600 where comid in (13294304, 13294298, 13294300, 13294310);
```
Test requests can be run using upstream or downstream features. These should end when reaching the Lake Mendota area. Making a request from a coastal feature (localhost:8080/nldi/linked-data/wqp/WIDNR_WQX-10031720/navigation/UT/flowlines?distance=100&f=json) should return only that feature.

Resolves #39 

After making a pull request
---------------------------
- [x] If appropriate, put the link to the PR in the JIRA ticket
- [x] Assign someone to review unless the change is trivial

### CODE REVIEW

**Review by mteather-usgs — APPROVED** [review]
The updates look good.  At some point we should add a unit test to cover this new coastal case, but that can be done in the near future.  Not necessary to hold up the release in my opinion.

**Comment by dblodgett-usgs** [comment]
I'd like to get the test case included before merging. Let's avoid that technical debt if possible.

File: src/main/resources/mybatis/navigate.xml
Comment (dblodgett-usgs) [inline src/main/resources/mybatis/navigate.xml:156]: This is kind of a magic number. Can we create a variable called "coastal_fcode = 56600" then use that variable? And/or please comment this line describing what this is doing.

Comment (mteather-usgs) [reply src/main/resources/mybatis/navigate.xml:156]: Oh, good catch.  That should probably be a constant, or even a configuration setting if we think it may change frequently.

Ethan, I know you ran into some challenges setting up the unit test, but we can figure it out together.

Comment (dblodgett-usgs) [reply src/main/resources/mybatis/navigate.xml:156]: It won't change. I'm ok with it being hard coded but it should be given an expressive variable name.

**Comment by EthanGrahn** [comment]
@dblodgett-usgs Moved the fcode to a variable in the latest commit. https://github.com/ACWI-SSWD/nldi-services/pull/203/commits/355f8ca3bcc2701ad41ff71ccc42adf76c30b52f

**Review by dblodgett-usgs — APPROVED** [review]

**Pull request merged**
